pkg/history/history.go

```diff
@@ -0,0 +1,409 @@
+func loadLogs(limit int) ([]CommandLog, error) {
+	var logs []CommandLog
+
+	logPath, err := getLogPath()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get log path: %w", err)
+	}
+
+	data, err := os.ReadFile(logPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read log file: %w", err)
+	}
+
+	if err := json.Unmarshal(data, &logs); err != nil {
+		return nil, fmt.Errorf("failed to parse log file: %w", err)
+	}
+
+	// Apply limit if specified
+	if limit > 0 && len(logs) > limit {
+		logs = logs[len(logs)-limit:]
+	}
+
+	return logs, nil
+}
+
+// saveLogs writes command history entries to disk
+// Enforces maximum entry limit
+func saveLogs(logs []CommandLog) error {
+	if len(logs) > maxEntries {
+		logs = logs[len(logs)-maxEntries:]
+	}
+
+	logPath, err := getLogPath()
+	if err != nil {
+		return fmt.Errorf("failed to get log path: %w", err)
+	}
+
+	data, err := json.MarshalIndent(logs, "", "  ")
+	if err != nil {
+		return fmt.Errorf("failed to marshal logs: %w", err)
+	}
+
+	if err := os.WriteFile(logPath, data, 0644); err != nil {
+		return fmt.Errorf("failed to write log file: %w", err)
+	}
+
+	return nil
+}
+
+// deriveCommandType determines the type of command being executed
+// Returns uppercase string representation of command hierarchy
+func deriveCommandType(cmd *cobra.Command) string {
+	path := cmd.CommandPath()
+	parts := strings.Fields(path)
+	if len(parts) <= 1 {
+		return "ROOT"
+	}
+	parts = parts[1:]
+	return strings.ToUpper(strings.Join(parts, "_"))
+}
+
+// appendLog adds a new command entry to the history
+func appendLog(entry CommandLog) error {
+	logs, err := loadLogs(0)
+	if err != nil {
+		logs = []CommandLog{}
+	}
+
+	logs = append(logs, entry)
+	return saveLogs(logs)
+}
+
+// wrapCommand wraps a cobra command to capture its execution details
+// Handles command interruption, output capture, and error logging
+func wrapCommand(cmd *cobra.Command) {
+	// Setup signal handling for interrupts
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
+
+	// Store original command functions
+	originalHelpFunc := cmd.HelpFunc()
+	originalPreRun := cmd.PreRun
+	originalPreRunE := cmd.PreRunE
+
+	// Wrap help function to capture help command execution
+	cmd.SetHelpFunc(func(cmd *cobra.Command, args []string) {
+		var stdout, stderr bytes.Buffer
+		oldOut := cmd.OutOrStdout()
+		oldErr := cmd.ErrOrStderr()
+
+		// Setup output capturing
+		cmd.SetOut(io.MultiWriter(oldOut, &stdout))
+		cmd.SetErr(io.MultiWriter(oldErr, &stderr))
+
+		// Run help in goroutine to handle interrupts
+		done := make(chan bool, 1)
+		go func() {
+			originalHelpFunc(cmd, args)
+			done <- true
+		}()
+
+		// Wait for completion or interrupt
+		select {
+		case <-done:
+		case <-sigChan:
+		}
+
+		// Log help command execution
+		logEntry := CommandLog{
+			Date:        time.Now(),
+			Input:       strings.Join(os.Args, " "),
+			Output:      stdout.String(),
```

Comment:
Do we know why some commands end up with nil output?:

```json
{
  "date": "2024-11-04T16:02:37.962884629Z",
  "input": "./corectl app create js-app --log-level debug --dry-run",
  "exitCode": 0,
  "command_type": "APPS_CREATE"
},
{
  "date": "2024-11-04T16:40:31.492094867Z",
  "input": "./corectl update",
  "exitCode": 130,
  "command_type": "UPDATE"
}
```

Reply:
Interactive commands I believe, will take a look if theres anything i can do